lib/engine/game/g_1856/game.rb

```diff
@@ -1598,12 +1601,17 @@ def national_token_swap
 
           # Then reduce down to limit
           # TODO: Possibly override ReduceTokens?
-          if national.tokens.size > national_token_limit
+          tokens_to_keep = [home_bases.size, national_token_limit].max
+          if national.tokens.size > tokens_to_keep
             @log << "#{national.name} will is above token limit and must decide which tokens to remove"
-            # TODO: implement this case, maybe use a varaiation of the below?
-            # @round.corporations_removing_tokens = [buyer, acquired_corp]
+            # This will be resolved in RemoveTokens
+            @round.pending_removals << {
+              corp: national,
+              count: national.tokens.size - tokens_to_keep,
+              hexes: national.tokens.reject { |t| home_bases.include?(t.hex) }.map(&:hex),
```

Comment:
tokens.map(&:hex).reject { |hex| home_bases.include?(hex) }

Reply:
This is updated now